### PR TITLE
[POAE7-1434] Developer Guide

### PR DESCRIPTION
**Pull request by sijialouintel**
added Developer Guide

### CODE REVIEW

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+cd build
+cmake ..
+make
+```
+
+_Notes: 
+To build without GPU support you can pass `-DENABLE_CUDA=off` to CMake. To enable embedded OmniSci build, you should pass `-DENABLE_DBE=on`. Conda build has CUDA disabled and embedded OmniSci enabled by default._
+
+#### <a name="1.2.3"></a>1.2.3. Build with docker image
+You can download docker image from this [link](http://sr129/file/package/docker_images/omnisci-dev.tar), which has some dependencies pre-installed. The purpose of using docker is to reduce environment issues. You can modify code in host and do build/debug/run jobs inside docker. (The docker image is about 7GB, conda install a lot dependency. And this image is internal only, I upload proxy info inside the image. Please re-build an image if want use it in CI tools like github action.)
```
Comment (winningsix) [inline DEVELOPER_GUIDE.md:85]: This link is not available outside company. I prefer to remove docker image approach here replacing with dockerfile one. Your thoughts?

**Review by winningsix — COMMENTED** [review]
LGTM.

File: DEVELOPER_GUIDE.md (hunk 1 above)
Comment (sijialouintel) [reply DEVELOPER_GUIDE.md:85]: Sure. Will update this after finishing dockerfile.

**Comment by sijialouintel** [comment]
should be merged after https://github.com/Intel-bigdata/omniscidb/pull/46 and https://github.com/Intel-bigdata/omniscidb/pull/47

File: DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,501 @@
...
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
+    "configurations": [
+     {
+      "name": "PlanConverterTest",    //??
+      "type": "cppdbg",
+      "request": "launch",
+      "program": "/home/xdy/projects/presto_cpp/build/presto_cpp/presto_protocol/tests/presto_protocol_test",
```
Comment (winningsix) [inline DEVELOPER_GUIDE.md:215]: let's hide internal path /foo/bar/

**Review by winningsix — APPROVED** [review]
Minor comments left.

File: DEVELOPER_GUIDE.md (hunk 2 above)
Comment (sijialouintel) [reply DEVELOPER_GUIDE.md]: finished modification

**Pull request merged**